From 82455b2826889525cfb9236c3c9d79cbf6460bcf Mon Sep 17 00:00:00 2001
From: Mikhail Novosyolov <m.novosyolov@rosalinux.ru>
Date: Sun, 26 Jan 2020 16:37:34 +0300
Subject: [PATCH] Fix linkage of popen_deadlock test

DEBUG: BUILDSTDERR: /usr/bin/ld: infinite_loop_dlsym.o: in function `native_loop_dlsym':
DEBUG: BUILDSTDERR: /builddir/build/BUILD/ruby-2.7.0/ext/-test-/popen_deadlock/infinite_loop_dlsym.c:16: undefined reference to `dlsym'
DEBUG: BUILDSTDERR: collect2: error: ld returned 1 exit status

Ruby was built with LibreSSL.
---
 ext/-test-/popen_deadlock/extconf.rb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/-test-/popen_deadlock/extconf.rb b/ext/-test-/popen_deadlock/extconf.rb
index 24a7d79931..b966fb7b32 100644
--- a/ext/-test-/popen_deadlock/extconf.rb
+++ b/ext/-test-/popen_deadlock/extconf.rb
@@ -1,5 +1,6 @@
 # frozen_string_literal: false
 case RUBY_PLATFORM
 when /solaris/i, /linux/i
+  $LDFLAGS << " -ldl"
   create_makefile("-test-/popen_deadlock/infinite_loop_dlsym")
 end
-- 
2.20.1

